assets/src/components/dummyDetourPage.tsx

```diff
@@ -16,19 +45,60 @@ export const DummyDetourPage = () => {
   }, [])
   const route = useRoute(routePattern?.routeId)
 
+  // The state machine won't "reinitialize", if it's inputs change, so only
+  // compute the snapshot from `localStorage` once.
+  const snapshot = useMemo(() => {
+    const snapshot_string = localStorage.getItem("snapshot")
+
+    if (!snapshot_string) {
+      return Err<"No Snapshot Present">("No Snapshot Present")
+    }
+
+    let persisted_snapshot
+    try {
+      persisted_snapshot = JSON.parse(snapshot_string)
+    } catch (error) {
+      return Err<"Parsing JSON Failed">("Parsing JSON Failed")
+    }
+
+    return isValidSnapshot(createDetourMachine, persisted_snapshot)
+  }, [])
+
   return (
     <>
-      {route && routePattern && routePattern.shape && (
+      {isOk(snapshot) ? (
         <DiversionPage
-          originalRoute={{
-            route,
-            routePattern,
-            center: { lat: 42.36, lng: -71.13 },
-            zoom: 16,
-          }}
+          snapshot={snapshot.ok}
           showConfirmCloseModal={false}
+          onClose={() => {
+            // If the close button is clicked,
+            // clear snapshot from storage and reinitialize state machine
+            localStorage.setItem("snapshot", "")
+            reload()
+          }}
         />
+      ) : (
+        <>
+          {
+            /* temp: since this is an internal page, show snapshot error as text
+             */
+            snapshot.err
+          }
+          {route && routePattern && routePattern.shape && (
+            <DiversionPage
+              originalRoute={{
+                center: { lat: 42.36, lng: -71.13 },
+                zoom: 16,
+                route,
+                routePattern,
+              }}
+              showConfirmCloseModal={false}
+              onClose={() => {}}
+            />
+          )}
+        </>
       )}
+      {}
```

Comment:
👀 What's this lil bracket doing on its lonesome?

Reply:
No idea! Thanks for catching!